lib/lib-search/src/main/java/org/sagebionetworks/search/awscloudsearch/SynapseCreatedCloudSearchField.java

```diff
@@ -0,0 +1,95 @@
+package org.sagebionetworks.search.awscloudsearch;
+
+import com.amazonaws.services.cloudsearchv2.model.IndexField;
+import com.amazonaws.services.cloudsearchv2.model.IndexFieldType;
+import org.sagebionetworks.util.ValidateArgument;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Wrapper for IndexFields defined in CloudSearch.
+ */
+class SynapseCreatedCloudSearchField implements CloudSearchField{
+	private static final Map<IndexFieldType, Function<IndexField, ?>> INDEX_OPTIONS_GETTER_MAP;
+	static{
+		Map<IndexFieldType, Function<IndexField, ?>> temp = new EnumMap<>(IndexFieldType.class);
+		temp.put(IndexFieldType.Literal, IndexField::getLiteralOptions);
+		temp.put(IndexFieldType.LiteralArray, IndexField::getLiteralArrayOptions);
+		temp.put(IndexFieldType.Text, IndexField::getTextOptions);
+		temp.put(IndexFieldType.TextArray, IndexField::getTextArrayOptions);
+		temp.put(IndexFieldType.Int, IndexField::getIntOptions);
+		temp.put(IndexFieldType.IntArray, IndexField::getIntArrayOptions);
+		temp.put(IndexFieldType.Date, IndexField::getDateOptions);
+		temp.put(IndexFieldType.DateArray, IndexField::getDateArrayOptions);
+		temp.put(IndexFieldType.Latlon, IndexField::getLatLonOptions);
+		temp.put(IndexFieldType.Double, IndexField::getDoubleOptions);
+		temp.put(IndexFieldType.DoubleArray, IndexField::getDoubleArrayOptions);
+		INDEX_OPTIONS_GETTER_MAP = Collections.unmodifiableMap(temp);
+	}
+
+
+	private IndexField indexField;
```

Comment:
add fields for searchable, faceted, and returned. use reflection in constructor to retrieve the values from the indexField and then set these fields. We don't expect the indexField to be modified at runtime so there is no reason to re-do the Reflection every time the getters are called.